### ISSUE TITLE
Trouble understanding toHaveStyleRule()

### ISSUE THREAD

**Issue by AshCoolman** [styled-components/jest-styled-components#23]
Hi, 
I'm trying to use `toHaveStyleRule` but I'm unsure how to use it. My naive modification to `jest-styled-components/test/index.spec.js` is here:

```js
test('react-test-renderer', () => {
  const tree = renderer.create(
    <Wrapper>
      <Title>Hello World, this is my first styled component!</Title>
    </Wrapper>,
  ).toJSON()

  expect(tree).toMatchStyledComponentsSnapshot()
  expect(tree).toHaveStyleRule('color', 'papayawhip');
})

test('enzyme', () => {
  const tree = shallow(
    <Wrapper>
      <Title>Hello World, this is my first styled component!</Title>
    </Wrapper>,
  )

  expect(tree).toMatchStyledComponentsSnapshot();
  expect(tree).toHaveStyleRule('color', 'papayawhip');
})

```

But it throws errors:

```sh
 FAIL  test/index.spec.js
  ● react-test-renderer


            "Expected [object Object] to be a component from react-test-renderer, or a mounted enzyme component."

            "But had an error, TypeError: Cannot read property 'styleSheet' of undefined"


      at Object.<anonymous> (test/index.spec.js:45:16)
      at process._tickCallback (internal/process/next_tick.js:103:7)

  ● enzyme


            "Expected [object Object] to be a component from react-test-renderer, or a mounted enzyme component."

            "But had an error, TypeError: Cannot read property '_renderedComponent' of undefined"


      at Object.<anonymous> (test/index.spec.js:56:16)
      at process._tickCallback (internal/process/next_tick.js:103:7)

  ✓ snapshot on null (8ms)
  ✕ react-test-renderer (47ms)
  ✕ enzyme (13ms)
```

Any help would be great, Thanks

**Comment by MicheleBertoli** [comment]
Hello @AshCoolman, thanks for opening this issue.
Which version of `styled-components` are you using?
We are having a [similar](https://github.com/styled-components/jest-styled-components/pull/14#issuecomment-304196142) issue with v2, which is going to be fixed soon.

**Comment by AshCoolman** [comment]
Ah yes, I am on v2 also

```js
"styled-components": "2.0.0"
```

```sh
$ node_modules/.bin/jest --version
v20.0.4
```

**Comment by ParAnton** [comment]
I'm having the same issue in toHavestyleRule.js file the const styleSheet is undefined since it doesn't seem to be declared in styled-components
`const { styleSheet } = require('styled-components')`

**Comment by MicheleBertoli** [comment]
Fixed by #25
